core/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -73,15 +68,13 @@
 /**
  * Context object used to create lucene queries on the shard level.
  */
-public class QueryShardContext {
+public class QueryShardContext extends QueryRewriteContext {
```

Comment:
the obvious thing. it's a true subclass of QueryRewriteContext it shares all it's members it's just more powerful. If you have a `QueryShardContext` there is no need to create another instance you can just pass it on?